src/func.rs

```diff
@@ -407,7 +407,11 @@ impl Func {
     ///
     /// let query = Query::select()
     ///     .expr(Func::coalesce([
-    ///         Expr::col(Char::SizeW).into(),
+    ///         Query::select()
+    ///             .from(Char::Table)
+    ///             .expr(Func::max(Expr::col(Char::SizeW)))
+    ///             .to_owned()
+    ///             .into(),
```

Comment:
I would rather be more conservative on this type conversion. Would a method `into_sub_query_expr` with an explicit intent better for this use case?

Reply:
Sorry for the slow reply.
Yes, perhaps `into_sub_query_expr` would be better (perhaps even better something like `into_scalar_subquery_expr` and implementing a separate `ScalarSubQueryStatement`)

But doesn't it create confusion that for `coalesce` SelectStatement needs to be explicitly converted to `SubQueryStatement`, and all other functions need to pass `SelectStatement` when called

Seems like this would require redoing the whole api or am I complicating things and it's simpler?